### ISSUE TITLE
missing scan configs

### ISSUE THREAD

**Issue by alexmateescu** [Atomicorp/gvm#25]
used the readme to install on centos 8 but when trying to create a task i get 
"Failed to find config 'daba56c8-73ec-11df-a475-002264764cea' ".
looking in /var/lib/gvm/data-objects i see the files in there. see below. 
what could be wrong?

/var/lib/gvm/data-objects
/var/lib/gvm/data-objects/gvmd
/var/lib/gvm/data-objects/gvmd/20.08
/var/lib/gvm/data-objects/gvmd/21.04
/var/lib/gvm/data-objects/gvmd/feed.xml
/var/lib/gvm/data-objects/gvmd/sha256sums
/var/lib/gvm/data-objects/gvmd/sha256sums.asc
/var/lib/gvm/data-objects/gvmd/timestamp
/var/lib/gvm/data-objects/gvmd/20.08/configs
/var/lib/gvm/data-objects/gvmd/20.08/port_lists
/var/lib/gvm/data-objects/gvmd/20.08/report_formats
/var/lib/gvm/data-objects/gvmd/20.08/configs/base-d21f6c81-2b88-4ac1-b7b4-a2a9f2ad4663.xml
/var/lib/gvm/data-objects/gvmd/20.08/configs/discovery-8715c877-47a0-438d-98a3-27c7a6ab2196.xml
/var/lib/gvm/data-objects/gvmd/20.08/configs/empty-085569ce-73ed-11df-83c3-002264764cea.xml
/var/lib/gvm/data-objects/gvmd/20.08/configs/full-and-fast-daba56c8-73ec-11df-a475-002264764cea.xml
/var/lib/gvm/data-objects/gvmd/20.08/configs/host-discovery-2d3f051c-55ba-11e3-bf43-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/20.08/configs/policy-it-grundschutz-c4b7c0cb-6502-4809-b034-8e635311b3e6.xml
/var/lib/gvm/data-objects/gvmd/20.08/configs/system-discovery-bbca7412-a950-11e3-9109-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/20.08/port_lists/all-iana-assigned-tcp-33d0cd82-57c6-11e1-8ed1-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/20.08/port_lists/all-iana-assigned-tcp-and-udp-4a4717fe-57d2-11e1-9a26-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/20.08/port_lists/all-tcp-and-nmap-top-100-udp-730ef368-57e2-11e1-a90f-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/20.08/report_formats/anonymous-xml-5057e5cc-b825-11e4-9d0e-28d24461215b.xml
/var/lib/gvm/data-objects/gvmd/20.08/report_formats/csv-results-c1645568-627a-11e3-a660-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/20.08/report_formats/itg-77bd6c4a-1f62-11e1-abf0-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/20.08/report_formats/pdf-c402cc3e-b531-11e1-9163-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/20.08/report_formats/txt-a3810a62-1f62-11e1-9219-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/20.08/report_formats/xml-a994b278-1f62-11e1-96ac-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/21.04/configs
/var/lib/gvm/data-objects/gvmd/21.04/port_lists
/var/lib/gvm/data-objects/gvmd/21.04/report_formats
/var/lib/gvm/data-objects/gvmd/21.04/configs/base-d21f6c81-2b88-4ac1-b7b4-a2a9f2ad4663.xml
/var/lib/gvm/data-objects/gvmd/21.04/configs/discovery-8715c877-47a0-438d-98a3-27c7a6ab2196.xml
/var/lib/gvm/data-objects/gvmd/21.04/configs/empty-085569ce-73ed-11df-83c3-002264764cea.xml
/var/lib/gvm/data-objects/gvmd/21.04/configs/full-and-fast-daba56c8-73ec-11df-a475-002264764cea.xml
/var/lib/gvm/data-objects/gvmd/21.04/configs/host-discovery-2d3f051c-55ba-11e3-bf43-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/21.04/configs/policy-it-grundschutz-c4b7c0cb-6502-4809-b034-8e635311b3e6.xml
/var/lib/gvm/data-objects/gvmd/21.04/configs/system-discovery-bbca7412-a950-11e3-9109-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/21.04/port_lists/all-iana-assigned-tcp-33d0cd82-57c6-11e1-8ed1-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/21.04/port_lists/all-iana-assigned-tcp-and-udp-4a4717fe-57d2-11e1-9a26-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/21.04/port_lists/all-tcp-and-nmap-top-100-udp-730ef368-57e2-11e1-a90f-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/21.04/report_formats/anonymous-xml-5057e5cc-b825-11e4-9d0e-28d24461215b.xml
/var/lib/gvm/data-objects/gvmd/21.04/report_formats/csv-results-c1645568-627a-11e3-a660-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/21.04/report_formats/itg-77bd6c4a-1f62-11e1-abf0-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/21.04/report_formats/pdf-c402cc3e-b531-11e1-9163-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/21.04/report_formats/txt-a3810a62-1f62-11e1-9219-406186ea4fc5.xml
/var/lib/gvm/data-objects/gvmd/21.04/report_formats/xml-a994b278-1f62-11e1-96ac-406186ea4fc5.xml

**Comment by presianbg** [comment]
Hi,

I'm having the same issue. Looks like that `gvmd` is crashing while loading the `NVTs` : 


```
Aug 26 20:47:25 openvas20 systemd-coredump[5322]: Process 3711 (gvmd) of user 992 dumped core.#012#012Stack trace of thread 3711:#012#0  0x00007f6db75ac82a __GI___strlen_sse2 (libc.so.6)#012#1  0x000055c912f82abd sql_quote (gvmd)#012#2  0x000055c912fcf80a set_nvts_feed_version (gvmd)#012#3  0x000055c912fd0731 update_nvts_from_vts (gvmd)#012#4  0x000055c912fd1a21 manage_update_nvt_cache_osp (gvmd)#012#5  0x000055c912f70678 update_nvt_cache_osp (gvmd)#012#6  0x000055c912f80ef8 manage_sync (gvmd)#012#7  0x000055c912f7218a serve_and_schedule (gvmd)#012#8  0x00007f6db75366a3 __libc_start_main (libc.so.6)#012#9  0x000055c912f6f0be _start (gvmd)
```
This is happening on up-to-date centos 8 minimal install.

In this latest Greenbone version Scan Config are being loaded after NVTs :
```
Scan configs will not be imported without synced VTs. gvmd needs to be connected to an ospd-openvas instance and VTs must be available on the hard disk at <install-prefix>/var/lib/openvas/plugins after running greenbone-nvt-sync.
```
Source - https://community.greenbone.net/t/gvm-20-08-stable-initial-release-2020-08-12/6312

but since gvmd is stuck on this stage:
```
gvm         5330  0.0  0.0 228756  2032 ?        S    20:47   0:00 gvmd: Reloading NVTs
gvm         5332 33.9  6.5 757604 533324 ?       S    20:47   1:32 gvmd: OSP: Updating NVT cache
```
We can't see the configs.

I guess something is wrong with the gvmd binary, but failed to find what.


EDIT: Just checked and this is what the coredump info shows:
```
        Signal: 11 (SEGV)
     Timestamp: Wed 2020-08-26 20:47:20 EEST (10min ago)
  Command Line: gvmd: OSP: Updating NVT cache
    Executable: /usr/sbin/gvmd
 Control Group: /system.slice/gvmd.service
          Unit: gvmd.service
         Slice: system.slice
       Boot ID: 561ef9b2a4aa451c8e16195f3c173a94
    Machine ID: 6a812faf1b6446d4aeb75bc696274d95
      Hostname: openvas20
       Storage: /var/lib/systemd/coredump/core.gvmd.992.561ef9b2a4aa451c8e16195f3c173a94.3711.1598464040000000.lz4
       Message: Process 3711 (gvmd) of user 992 dumped core.
                
                Stack trace of thread 3711:
                #0  0x00007f6db75ac82a __GI___strlen_sse2 (libc.so.6)
                #1  0x000055c912f82abd sql_quote (gvmd)
                #2  0x000055c912fcf80a set_nvts_feed_version (gvmd)
                #3  0x000055c912fd0731 update_nvts_from_vts (gvmd)
                #4  0x000055c912fd1a21 manage_update_nvt_cache_osp (gvmd)
                #5  0x000055c912f70678 update_nvt_cache_osp (gvmd)
                #6  0x000055c912f80ef8 manage_sync (gvmd)
                #7  0x000055c912f7218a serve_and_schedule (gvmd)
                #8  0x00007f6db75366a3 __libc_start_main (libc.so.6)
                #9  0x000055c912f6f0be _start (gvmd)
```

**Comment by atomicturtle** [comment]
When you were setting this up, did you use the default "admin" username, or choose your own?

**Comment by presianbg** [comment]
I have used the default `admin` . Is that matter?

**Comment by atomicturtle** [comment]
It could have mattered if it was a different user. This GVM update adds a feed user type, and you have to assign that user to it, that is what could cause  "Failed to find config 'daba56c8-73ec-11df-a475-002264764cea' ".  I think yours might be something completely different though, note how its referencing the NVT's in your message. How many do you have in /var/lib/gvm/plugins?

**Comment by presianbg** [comment]
the part with `Failed to find config ... ` is because NVTs are not completely loaded and thus the configs are not synced :

`Scan configs will not be imported without synced VTs. gvmd needs to be connected to an ospd-openvas instance and VTs must be available on the hard disk at <install-prefix>/var/lib/openvas/plugins after running greenbone-nvt-sync.`

In ` /var/lib/gvm/plugins` I have `61785`  files (359MB).

**Comment by atomicturtle** [comment]
Yeah that part is pretty tough to handle, the rsync mirror isnt always available. I tried adding in some tolerance to that, but clearly we need to make that tougher. Did you catch how many you had? I suppose I could change the loop to keep retrying until the count was over 60k

**Comment by presianbg** [comment]
@atomicturtle  I'm not sure what are you asking for. Would you be so kind to elaborate on what information you need?
As far as I understood, you think that the coredump crash is due downloading (rsyncing) the NVTs is taking too long ?

**Comment by atomicturtle** [comment]
Actually lets start with a unit test, what do we need to do to repeat the condition you're running in to?

**Comment by alexmateescu** [comment]
brand new install of the atomic packages should do it 
i can start a new vm and try to install to see what's happening

**Comment by presianbg** [comment]
@alexmateescu  I already tried to install a new VM - Centos 8 minimal install (all up-to-date). I'm using Hyper-V and the VM has 4 CPU cores and 8GB RAM. 
Then disabled selinux and installed gvm from the repo. 
After this started the gvm-setup as root and used `admin` with some random password. 
Result is the same as my previous test - every 10 minutes `gvmd` crashes with coredump while `Updating NVT cache`

**Comment by alexmateescu** [comment]
@presianbg i understand but let's just do it one more time. i am in the same boat. if i build it from sources it works

**Comment by alexmateescu** [comment]
@atomicturtle I was chatting with the guys in GVM(Openvas) about this issue and they were saying that apparently your setup script does not set a feed owner correctly. and because the scan configs come from the feed now rather than being embedded with the software it does not populate them. this might be why the NVTs don't update.

**Comment by alexmateescu** [comment]
@presianbg did you ran the setup script as root? i just remembered that on the source installation i had to run the commands as "gvm" user and i wonder if this is not the case here

**Comment by atomicturtle** [comment]
Hmm, so this is what is used to set the feed owner right now:

`        su - gvm -c "/usr/sbin/gvmd --modify-setting 78eceaec-3385-11ea-b237-28d24461215b --value $FEED_OWNER"
`
Anyone know what that should be? I got that value from the README.md

**Comment by alexmateescu** [comment]
@atomicturtle i am not sure if that is the correct UUID for the setting 
https://community.greenbone.net/t/full-gvm-11-build-guide-for-centos-8/5425 this guide works for the source build but i can't find any setting of the feed_owner in there

**Comment by atomicturtle** [comment]
Yeah thats for v11, we're working with v20.08 here. The feed user was introduced in this version

**Comment by alexmateescu** [comment]
@atomicturtle 
gvmd --modify-setting 78eceaec-3385-11ea-b237-28d24461215b --value $(gvmd --get-users --verbose | grep psadmin | awk {‘print $2’})

CFI's answer from Greenbone
**_This is the correct UUID and AFAIK doesn’t change (i have three different setups using the same UUID)._**

**_Note that the import of the scan config might still fail as the gvmd binary provided via the atomic repository seems to be segfaulting for at least one user: https://github.com/Atomicorp/openvas/issues/25#issuecomment-681031197_**

**Comment by alexmateescu** [comment]
@atomicturtle so after more debugging it looks to me that it is not connecting to the scanner
_gvmd.log_
**md manage:   INFO:2020-09-01 15h48.48 utc:14209: OSP service has different VT status (version (null)) from database (version (null), 0 VTs). Starting update ...**
_openvas.log_
l**ib  nvticache:MESSAGE:2020-09-01 14h17.43 utc:11712: Updated NVT cache from version 0 to 202009011207**

**Comment by atomicturtle** [comment]
OK I see where it happens now, the coredump while not good, isnt related. Its being caused by the time gvmd takes to initialize (20 minutes on my test). So gvm-setup needs to watch the gvmd.log for it to report that it has finished, the closest log event I can find that indicates that is something like:


`
report_format:MESSAGE:2020-09-01 17h26.42 utc:5240: Report format XML (a994b278-1f62-11e1-96ac-406186ea4fc5) has been created by admin`

This has to finish before assigning the feed owner

**Comment by presianbg** [comment]
@atomicturtle I'm not sure how to handle this? Is it issue caused by my (our) VM, or the `setup` script needs to be adjusted ?

**Comment by atomicturtle** [comment]
right now, dont do anything at the admin setup prompt until you see it say its created that report format in the gvmd.log. This is a race condition, if you try to make it do something before that its going to error out

**Comment by micoots** [comment]
I've spent the better part of 2 days on this now, trying to debug etc. I built (in Azure) two CentOS 8.2 Gen 1 machines to test installs. I used the simple: https://github.com/Atomicorp/openvas instructions on the base system:

CentOS Linux release 8.2.2004 (Core)

Trying to create a task get that exact error first comment of this post.

The Secinfo -> NVTs are always empty with "No NVTs available".

I've read each post here, but unfortunately there doesn't seem to be any fix, or is there?

_right now, **dont do anything at the admin setup prompt** until you see it say its created that report format in the gvmd.log. This is a race condition, if you try to make it do something before that its going to error out_

What do you mean by "don't do anything at the admin setup prompt" ?

Do you mean that after we run the last step of item 3 "gvm-setup" and it reaches:

_Enter administrator username [Default: admin] :
Enter Administrator Password:
Verify Administrator Password:_

we need to monitor gvmd.log and see a message simiar to:

_report_format:MESSAGE:2020-09-01 17h26.42 utc:5240: Report format XML (a994b278-1f62-11e1-96ac-406186ea4fc5) has been created by admin_

then we continue with the admin account setup?

Please clarify thank you.

**Comment by micoots** [comment]
BTW.. I restarted a new build and after gvm-setup I am here now:

```
GVMD startup: Complete

Step 2: Choose the GSAD admin users password.
The admin user is used to configure accounts,
Update NVT's manually, and manage roles.

Enter administrator username [Default: admin] :
```

on one sesssion leaving it there, while the gvmd.log shows:

```
md manage:   INFO:2020-09-04 02h38.57 utc:17056: Updating /var/lib/gvm/cert-data/CB-K16.xml
md manage:   INFO:2020-09-04 02h38.57 utc:17056: Updating /var/lib/gvm/cert-data/CB-K17.xml
md manage:   INFO:2020-09-04 02h38.57 utc:17056: Updating /var/lib/gvm/cert-data/CB-K18.xml
md manage:   INFO:2020-09-04 02h38.58 utc:17056: Updating /var/lib/gvm/cert-data/CB-K19.xml
md manage:   INFO:2020-09-04 02h38.58 utc:17056: Updating /var/lib/gvm/cert-data/CB-K20.xml
md manage:   INFO:2020-09-04 02h38.58 utc:17056: Updating Max CVSS for DFN-CERT
md manage:   INFO:2020-09-04 02h39.02 utc:17056: Updating DFN-CERT CVSS max succeeded.
md manage:   INFO:2020-09-04 02h39.02 utc:17056: Updating Max CVSS for CERT-Bund
md manage:   INFO:2020-09-04 02h39.04 utc:17056: Updating CERT-Bund CVSS max succeeded.
md manage:   INFO:2020-09-04 02h39.04 utc:17056: sync_cert: Updating CERT info succeeded.
md manage:   INFO:2020-09-04 02h41.53 utc:17057: OSP service has different VT status (version (null)) from database (version (null), 0 VTs). Starting update ...
md manage:   INFO:2020-09-04 02h49.47 utc:17186: OSP service has different VT status (version (null)) from database (version (null), 0 VTs). Starting update ...
md manage:   INFO:2020-09-04 02h57.19 utc:17294: OSP service has different VT status (version (null)) from database (version (null), 0 VTs). Starting update ...
md manage:   INFO:2020-09-04 03h05.02 utc:17421: OSP service has different VT status (version (null)) from database (version (null), 0 VTs). Starting update ...
md manage:   INFO:2020-09-04 03h12.42 utc:17541: OSP service has different VT status (version (null)) from database (version (null), 0 VTs). Starting update ...
```

The other file in /var/log/gvm, namely openvas.log, shows:

`lib  nvticache:MESSAGE:2020-09-04 01h25.36 utc:11921: Updated NVT cache from version 0 to 202009030952`

Do I keep waiting at this point? or I press ENTER for "Enter administator username" ?

It's been a couple of hours waiting to get to this point.

**Comment by alexmateescu** [comment]
@micoots 
On this line OSP service has different VT status (version (null)) from database (version (null), 0 VTs). Starting update ...
the OSP service should report the version number from the openvas.log. that is where the problem is. I think it does not talk correctly with the scanner or something is wrong in there

Furthermore i am looking at the ospd-openvas.log and i see an error about connecting to redis. can you confirm you have it in yours as well? I think it is not talking to REDIS and that is where the problem is.

also in the Install.md of openvas there is a refrence to the GVM user being allowed to run some commands as root(see below) 
I have added the user in visudo but no luck so far. maybe it needs to be added in the gvm-setup

**_Please note that although you can run openvas as a user without elevated privileges, it is recommended that you start openvas as root since a number of Network Vulnerability Tests (NVTs) require root privileges to perform certain operations like packet forgery. If you run openvas as a user without permission to perform these operations, your scan results are likely to be incomplete.
As openvas will be launched from an ospd-openvas process with sudo, the next configuration is required in the sudoers file:_**

**_sudo visudo
add this line to allow the user running ospd-openvas, to launch openvas with root permissions_**

**_<user> ALL = NOPASSWD: <install prefix>/sbin/openvas
If you set an install prefix, you have to update the path in the sudoers file too:_**

**_Defaults        secure_path=<existing paths...>:<install prefix>/sbin_**

**Comment by micoots** [comment]
OK.. so from here then I just pressed ENTER for admin and entered the password I wanted:

```
Enter administrator username [Default: admin] :
Enter Administrator Password:
Verify Administrator Password:

Created symlink /etc/systemd/system/multi-user.target.wants/ospd-openvas.service â /usr/lib/systemd/system/ospd-openvas.service.
Created symlink /etc/systemd/system/openvas-manager.service â /usr/lib/systemd/system/gvmd.service.
Created symlink /etc/systemd/system/multi-user.target.wants/gvmd.service â /usr/lib/systemd/system/gvmd.service.
Created symlink /etc/systemd/system/multi-user.target.wants/gsad.service â /usr/lib/systemd/system/gsad.service.
FirewallD is not running
FirewallD is not running
FirewallD is not running

Setup complete
```

which immediately logged the following:

```
md manage:   INFO:2020-09-04 05h37.30 utc:19906: OSP service has different VT status (version (null)) from database (version (null),
0 VTs). Starting update ...
md   main:MESSAGE:2020-09-04 05h38.18 utc:20011:    Greenbone Vulnerability Manager version 20.08.0 (DB revision 233)
md manage:   INFO:2020-09-04 05h38.18 utc:20011:    Creating user.
md   main:MESSAGE:2020-09-04 05h38.22 utc:20015:    Greenbone Vulnerability Manager version 20.08.0 (DB revision 233)
md manage:   INFO:2020-09-04 05h38.22 utc:20015:    Modifying user password.
md   main:MESSAGE:2020-09-04 05h38.24 utc:20061:    Greenbone Vulnerability Manager version 20.08.0 (DB revision 233)
md manage:   INFO:2020-09-04 05h38.24 utc:20061:    Getting users.
md   main:MESSAGE:2020-09-04 05h38.27 utc:20105:    Greenbone Vulnerability Manager version 20.08.0 (DB revision 233)
md manage:   INFO:2020-09-04 05h38.27 utc:20105:    Modifying setting.
event port_list:MESSAGE:2020-09-04 05h38.34 utc:16206: Port list All IANA assigned TCP (33d0cd82-57c6-11e1-8ed1-406186ea4fc5) has been created by admin
event port_list:MESSAGE:2020-09-04 05h38.35 utc:16206: Port list All IANA assigned TCP and UDP (4a4717fe-57d2-11e1-9a26-406186ea4fc5) has been created by admin
event port_list:MESSAGE:2020-09-04 05h38.35 utc:16206: Port list All TCP and Nmap top 100 UDP (730ef368-57e2-11e1-a90f-406186ea4fc5) has been created by admin
event report_format:MESSAGE:2020-09-04 05h38.35 utc:16206: Report format Anonymous XML (5057e5cc-b825-11e4-9d0e-28d24461215b) has been created by admin
event report_format:MESSAGE:2020-09-04 05h38.35 utc:16206: Report format CSV Results (c1645568-627a-11e3-a660-406186ea4fc5) has been created by admin
event report_format:MESSAGE:2020-09-04 05h38.35 utc:16206: Report format ITG (77bd6c4a-1f62-11e1-abf0-406186ea4fc5) has been created by admin
event report_format:MESSAGE:2020-09-04 05h38.35 utc:16206: Report format PDF (c402cc3e-b531-11e1-9163-406186ea4fc5) has been created by admin
event report_format:MESSAGE:2020-09-04 05h38.35 utc:16206: Report format TXT (a3810a62-1f62-11e1-9219-406186ea4fc5) has been created by admin
event report_format:MESSAGE:2020-09-04 05h38.35 utc:16206: Report format XML (a994b278-1f62-11e1-96ac-406186ea4fc5) has been created by admin
```

The daba56c8-73ec-11df-a475-002264764cea doesn't seem to be part of this?

I'm leaving it here at this stage, maybe for another hour, before I try accessing the GUI.

The only additional log in /var/log/gvm was gsad.log, which contains:

`gsad main:MESSAGE:2020-09-04 05h38.29 utc:20151: Starting GSAD version 20.08.0`

which indicates to me the GUI is active, but I'll wait any way.

Hmm.. after a few minutes the gvm.log still comes back with:

`md manage:   INFO:2020-09-04 05h44.53 utc:20305: OSP service has different VT status (version (null)) from database (version (null), 0 VTs). Starting update ...
`

**Comment by alexmateescu** [comment]
just for the fun of it i will try to build it from sources first on Ubuntu and then on Centos 8 to see if it works

**Comment by micoots** [comment]
@alexmateescu what documentation / step-by-step guide is there to build this from source? I haven't been able to find anything on GVM v20, only for v11 which is no good.

I don't mind doing this myself on CentOS 8 if you have step-by-step instructions, I'll just boot up another CentOS 8 VM in Azure and kick it off.

**Comment by alexmateescu** [comment]
Lookup openvas on github and you will find it

**Comment by max70-root** [comment]
Exactly same problem here: brand new installation on Centos 8 VM using atomic repo i have gvmd crash dump with

`Sep  4 16:40:57 localhost systemd-coredump[12314]: Process 12285 (gvmd) of user 992 dumped core.#012#012Stack trace of thread 12285:#012#0  0x00007fa357c2482a __GI___strlen_sse2 (libc.so.6)#012#1  0x000055d6e70f0abd sql_quote (gvmd)#012#2  0x000055d6e713d80a set_nvts_feed_version (gvmd)#012#3  0x000055d6e713e731 update_nvts_from_vts (gvmd)#012#4  0x000055d6e713fa21 manage_update_nvt_cache_osp (gvmd)#012#5  0x000055d6e70de678 update_nvt_cache_osp (gvmd)#012#6  0x000055d6e70eeef8 manage_sync (gvmd)#012#7  0x000055d6e70e018a serve_and_schedule (gvmd)#012#8  0x00007fa357bae6a3 __libc_start_main (libc.so.6)#012#9  0x000055d6e70dd0be _start (gvmd)`

and in gvmd.log i see

`md manage:   INFO:2020-09-04 14h48.20 utc:12730: OSP service has different VT status (version (null)) from database (version (null), 0 VTs). Starting update ...`


any fix suggested?

thank you

**Comment by atomicturtle** [comment]
So for what its worth, this core dump is not the issue (its a different, unrelated issue). This is a race condition. gvmd takes a long time to start up (10-15 minutes on my system), during which you'll get that coredump (again, its not the issue).  If you rush ahead of its startup before its finished setting everything up, that I believe is the root problem.  The setup script doesnt take this into account right now:


```
echo -n "GVMD startup: "
# TODO: Need some function that says its started
# the gvmd test does work, but only once the nvti's are going in logs
# this also needs to create the scan configs
echo "Complete"

```

So the workaround here is to watch gvmd.log to see if it has finished starting before trying to assign the feed user. A source build would have the same issue, but because it takes so much longer to do might avoid it because gvmd is running and doing its thing while you're building everything else. 

The fix would be to modify the code above with a way to detect that gvmd has completed its startup and initialization correctly.

**Comment by cfi-gb** [comment]
@atomicturtle While i think that the core dump / segfault and the missing scan configs are a effect of this problem the reason for it is a different one which is the following from the "Details section" of https://community.greenbone.net/t/gvm-20-08-stable-initial-release-2020-08-12/6312:

> The location of the feed-update.lock file for gvmd, openvas and ospd-openvas has been consolidated and all components expecting this file at the same location (which is <install-prefix>/var/run/feed-update.lock by default).
>
> Note: When changing the default by using a different OPENVAS_RUN_DIR and GVM_RUN_DIR (Especially important for package maintainers) please make sure to set the correct OPENVAS_FEED_LOCK_PATH and GVM_FEED_LOCK_PATH for cmake during build time

If i'm checking the ospd-openvas.service shipped with the OSPd-openvas package i can find the following:

```
--lock-file-dir /var/run/ospd
```

This is causing the ospd-openvas service to lock the file `/var/run/ospd/feed-update.lock` for the initial build of the NVT info / cache where the gvmd is looking for something like e.g.  `/var/run/gvm/feed-update.lock`.

And finally this difference in the usage of the feed-update.lock is causing gvmd to rebuild the scan config / NVTi cache too early way before ospd-openvas has finished its job.

To fix this changing `--lock-file-dir /var/run/ospd` to `--lock-file-dir /var/run/gvm` (or whatever "run dir" the packages are using) should solve this problem without having to monitor the gvmd.log for the final startup.

**Comment by winnerwq** [comment]
> To fix this changing `--lock-file-dir /var/run/ospd` to `--lock-file-dir /var/run/gvm` (or whatever "run dir" the packages are using) should solve this problem without having to monitor the gvmd.log for the final startup.

Hi cfi-gb, I tried to update  ospd-openvas.service, then reboot. But the problem persists.

**Comment by max70-root** [comment]
Hi
I tried the suggestion to fix the lock dir path but the problem is still there: gvmd constantly crashes and nvt feed does not update:

`Sep  7 09:56:09 localhost systemd-coredump[6342]: Removed old coredump core.gvmd.992.706ec5c41736447d9e0f5ed5f355a24a.53978.1599460019000000.lz4.
Sep  7 09:56:14 localhost systemd-coredump[6342]: Process 6322 (gvmd) of user 992 dumped core.#012#012Stack trace of thread 6322:#012#0  0x00007f53ea34d82a __GI___strlen_sse2 (libc.so.6)#012#1  0x0000563443526abd sql_quote (gvmd)#012#2  0x000056344357380a set_nvts_feed_version (gvmd)#012#3  0x0000563443574731 update_nvts_from_vts (gvmd)#012#4  0x0000563443575a21 manage_update_nvt_cache_osp (gvmd)#012#5  0x0000563443514678 update_nvt_cache_osp (gvmd)#012#6  0x0000563443524ef8 manage_sync (gvmd)#012#7  0x000056344351618a serve_and_schedule (gvmd)#012#8  0x00007f53ea2d76a3 __libc_start_main (libc.so.6)#012#9  0x00005634435130be _start (gvmd)
`

**Comment by cfi-gb** [comment]
1. Have you verified that the packages of this project are indeed using `/var/run/gvm` (a `feed-update.lock` should exist here if ospd-openvas isn't running)?
2. The setup might be already in an unrecoverable state, have you tried to start from scratch with a clean setup including 
3. Another issue reported recently in ospd-openvas: #30 could be related as well
4. This project might have changed `OPENVAS_RUN_DIR` so that part in https://github.com/Atomicorp/openvas/issues/25#issuecomment-687884201 could be relevant / required as well

Note that i'm not familiar with the GVM code base or the functionality of it related to the scan config import so the above is just some wild guessing.

**Comment by atomicturtle** [comment]
Updates  gvm-20.8.0-14790 and OSPd-openvas-20.8.0-14789 are out now. These would only address new installations, I don't believe there is any reliable mechanism to resolve this on an update. I'd recommend if you are experiencing issues with feed ownership, or segmentation faults to start over on a new image.

**Comment by max70-root** [comment]
With the new updates the problem is solved and my installation is now up and running. 
Thank you very much for your support.

**Comment by micoots** [comment]
I've also gone through and built two CentOS 8 using the new packages, and all worked well.

PDF reports however produce 0 byte files on download. To fix this, you need to do:

```
dnf -y install texlive-collection-fontsrecommended texlive-collection-latexrecommended texlive-changepage texlive-titlesec

mkdir -p /usr/share/texlive/texmf-local/tex/latex/comment

cd /usr/share/texlive/texmf-local/tex/latex/comment

wget http://mirrors.ctan.org/macros/latex/contrib/comment/comment.sty

chmod 644 comment.sty

texhash
```

You should add that into the setup instruction guide.

**Comment by atomicturtle** [comment]
Or just fix it like we did in the older GVM packages :P  Ill add that in next

**Comment by atomicturtle** [comment]
Closing this one out as solved, thanks for all the input everyone!

**Comment by willemdh** [comment]
Installed GVM-20 yesterday on clean CentOS 8 and also ran into this issue. Which is weird, as this is marked as resolved?
I did install as sudo ..., so i'll try the suggestion to install as root. Do I still have to wait at the admin pw prompt for an event like "report_format:MESSAGE:2020-09-01 17h26.42 utc:5240: Report format XML (a994b278-1f62-11e1-96ac-406186ea4fc5) has been created by admin"?

**Comment by willemdh** [comment]
I seem to have an issue with Postgres:

```
md manage:   INFO:2020-09-23 10h14.19 utc:24693: sync_cert: Updating CERT info succeeded.
md   main:MESSAGE:2020-09-23 10h22.12 utc:25639:    Greenbone Vulnerability Manager version 20.08.0 (DB revision 233)
md manage:   INFO:2020-09-23 10h22.12 utc:25639:    Getting users.
md manage:WARNING:2020-09-23 10h22.12 utc:25639: sql_open: PQconnectPoll failed
md manage:WARNING:2020-09-23 10h22.12 utc:25639: sql_open: PQerrorMessage (conn): FATAL:  role "root" does not exist
md manage:WARNING:2020-09-23 10h22.12 utc:25639: init_manage_process: sql_open failed
```

@atomicturtle Is Postgres config provided by the rpm or setup? if not which steps should I follow to corectly configure Postgres?

I can login as postgres and a list of users shows gvm as dba:

![image](https://user-images.githubusercontent.com/6462991/94001541-ed851680-fd98-11ea-87ce-93f3318b7b81.png)

**Comment by willemdh** [comment]
Ok, seem I shouldn't give gvm user sudo nopasswd privileges.. After retrying  a third time it worked. Tx

**Comment by cfi-gb** [comment]
> md manage:WARNING:2020-09-23 10h22.12 utc:25639: sql_open: PQerrorMessage (conn): FATAL:  role "root" does not exist

Note that this (AFAIK) means that you have started the `gvmd` service or any of it's terminal commands as the `root` user and not as the `gvm` user as used / defined / required by this project / package.
